### ISSUE TITLE
AttributeError: module 'email.errors' has no attribute 'InvalidHeaderError'.

### ISSUE THREAD

**Issue by The-Compiler** [python/cpython#93010]
**Bug report**

`email._header_value_parser` tries to raise `errors.InvalidHeaderError`, but that does not exist (and as far as I can tell, never did):

```python
>>> from email.headerregistry import HeaderRegistry
>>> reg = HeaderRegistry()
>>> reg('Content-Disposition', 'attachment; 0*00="foo"')
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/usr/lib/python3.10/email/headerregistry.py", line 604, in __call__
    return self[name](name, value)
  File "/usr/lib/python3.10/email/headerregistry.py", line 192, in __new__
    cls.parse(value, kwds)
  File "/usr/lib/python3.10/email/headerregistry.py", line 448, in parse
    kwds['parse_tree'] = parse_tree = cls.value_parser(value)
  File "/usr/lib/python3.10/email/_header_value_parser.py", line 2705, in parse_content_disposition_header
    disp_header.append(parse_mime_parameters(value[1:]))
  File "/usr/lib/python3.10/email/_header_value_parser.py", line 2569, in parse_mime_parameters
    token, value = get_parameter(value)
  File "/usr/lib/python3.10/email/_header_value_parser.py", line 2431, in get_parameter
    token, value = get_section(value)
  File "/usr/lib/python3.10/email/_header_value_parser.py", line 2382, in get_section
    section.defects.append(errors.InvalidHeaderError(
AttributeError: module 'email.errors' has no attribute 'InvalidHeaderError'. Did you mean: 'InvalidHeaderDefect'?
```

Code:

https://github.com/python/cpython/blob/d8537580921b2e02f477ff1a8dedcf82c24ef0c2/Lib/email/_header_value_parser.py#L2381-L2383

This was added in 97f43c019f3bb8376a3a5f7bc52e97b4c2ed8e16 by @bitdancer pretty much some 10 years ago, and I don't think it ever worked (but nor could I find anyone running into this so far): The git history doesn't know anything about an `email.errors.InvalidHeaderError`.

Found via [Hypothesis](https://hypothesis.readthedocs.io/en/latest/) while testing my own parsing code - cc @Zac-HD if you want to see another instance of "the curse of hypothesis" :wink:

**Your environment**

- CPython versions tested on: 3.7.13, 3.8.13, 3.9.13, 3.10.4, 3.11.0b1
- Operating system and architecture: Archlinux, x86_64

**Comment by warsaw** [comment]
Oh wow, nice find.

**Comment by Zac-HD** [comment]
Nice find!

On a meta-level though this suggests that CPython could really benefit from adopting some static analysis tooling and/or coverage requirements which are table stakes for any critical project... maybe I can argue for that instead of PBT at the next summit 😅

**Comment by The-Compiler** [comment]
> On a meta-level though this suggests that CPython could really benefit from adopting some static analysis tooling and/or coverage requirements which are table stakes for any critical project... maybe I can argue for that instead of PBT at the next summit

Agreed! FWIW I tried to run flake8 with a lot of ignored errors and manual processing, and I believe I now have another handful of bugs/PRs to report... :sweat_smile: 

```
Lib/_pyio.py:2165:28: F821 undefined name 'locale'
Lib/enum.py:1643:28: F821 undefined name 'cls_name'
Lib/unittest/test/test_case.py:85:38: F821 undefined name 'MyException'
Lib/unittest/test/test_case.py:106:38: F821 undefined name 'MyException'
Lib/unittest/test/test_case.py:117:38: F821 undefined name 'MyException'
Lib/unittest/test/test_suite.py:397:17: F821 undefined name 'fail'
Lib/test/test_shutil.py:1597:78: F821 undefined name 'x'
Lib/test/test_shutil.py:1599:78: F821 undefined name 'x'
Lib/test/test_shutil.py:1602:48: F821 undefined name 'x'
Lib/test/test_urllib2net.py:60:23: F821 undefined name 'ResourceDenied'
Lib/test/test_ssl.py:3485:21: F821 undefined name 'conn'
Lib/test/test_ssl.py:3486:31: F821 undefined name 'conn'
Lib/test/test_importlib/test_threaded_import.py:275:5: F821 undefined name 'unittets'
Lib/test/test_sqlite3/test_dbapi.py:312:13: F821 undefined name 'const'
Lib/test/test_sqlite3/test_dbapi.py:646:39: F821 undefined name 'size'

Lib/enum.py:483:30: F523 '...'.format(...) has unused arguments at position(s): 0
Lib/test/test_concurrent_futures.py:480:47: F522 '...'.format(...) has unused named argument(s): executor_type

Lib/encodings/mac_arabic.py:474:5: F601 dictionary key 32 repeated with different values
[...]
Lib/encodings/mac_arabic.py:593:5: F601 dictionary key 125 repeated with different values
Lib/test/test_mailbox.py:2265:5: F601 dictionary key 'Delivered-To' repeated with different values
Lib/test/test_mailbox.py:2266:5: F601 dictionary key 'Received' repeated with different values
[...]
```

**Comment by The-Compiler** [comment]
For anyone curious about the result of running `flake8` and taking a closer look at the result: I spent hours today reporting 9 more similar bugs, see them linked above. Decided to not report the `encodings.mac_arabic` one despite it probably being a genuine issue too - it was added in 2005, seems to be autogenerated, and hasn't seen any significant changes since then. Given how irrelevant it probably is today, I'm guessing nobody will care (and that's probably okay too!).
